Tangem/Modules/Send/StakingModel.swift

```diff
@@ -165,7 +165,20 @@ private extension StakingModel {
             return validateError
         }
 
-        return .readyToStake(.init(amount: newAmount, validator: validator, fee: fee, isFeeIncluded: includeFee))
+        let hasPreviousStakeOnSameValidator = stakingManager.state.balances?
+            .filter { $0.balanceType == .active }
+            .map(\.validatorAddress)
+            .contains(validator)
```

Comment:
как будто можно в один проход сделать, и читать проще будет условие )

```suggestion
        let hasPreviousStakeOnSameValidator = stakingManager.state.balances?.contains { balance in
            balance.balanceType == .active && balance.validatorAddress == validator
        } ?? false
```